modules/cli/src/main/scala/scala/cli/commands/bsp/Bsp.scala

```diff
@@ -114,6 +138,7 @@ object Bsp extends ScalaCommand[BspOptions] {
       val envs            = getEnvsFromFile()
       val bspBuildOptions = buildOptions(sharedOptions, launcherOptions, envs)
         .orElse(finalBuildOptions)
+      refreshPowerMode(launcherOptions, sharedOptions, envs)
```

Comment:
Why are you reloading that also twice before? Is it because the options change and at one of the previous stages `power` might have been set to true and you want to pick that up too?

Reply:
yes, exactly this. 
Whenever we load the latest `sharedOptions`, `launcherOptions` and `envs`, we also want to make sure we don't have to update power mode.